libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -483,11 +477,43 @@ private void updateProgressChangeListener() {
     }
   }
 
+  /**
+   * Updates the size of the bounding box used to determine which route line was clicked. Upon
+   * a map click an attempt will be made to find a route line intersecting the bounding box. This
+   * is used to select alternative route lines on the map.
+   *
+   * Passing large integers values through this method means that a user doesn't
+   * have to be very accurate with tapping on an actual route line because the invisible
+   * box is large and thus, there's a good chance that a portion of a route line will run
+   * through the query box's area. A smaller rectangle will decrease the chances.
+   */
+  public void updateClickDistancePadding(int newRouteClickRectFPadding) {
+    routeClickPadding = newRouteClickRectFPadding;
+  }
+
+  @Nullable
   @NonNull
   private MapRouteProgressChangeListener buildMapRouteProgressChangeListener() {
     return new MapRouteProgressChangeListener(routeLine, routeArrow);
   }
 
+  protected final MapboxMap.OnMapClickListener mapClickListener = new MapboxMap.OnMapClickListener() {
+    @Override
+    public boolean onMapClick(final @NonNull LatLng point) {
+      if (routeLine.retrieveVisibility() && routeLine.retrieveAlternativesVisible()) {
+        final int index = routeLine.findClosestRoute(point, mapboxMap, routeClickPadding);
+        if (index >= 0) {
```

Comment:
Should we return `true` if this check succeeds?

Reply:
If true is returned the map click is swallowed and can't be used for anything else.  The "legacy" version always returned false.

Reply:
We performed an action so we should consume the click. Otherwise, for example, there can be a scenario when one click of the map both switched the route and highlighted a POI which is confusing.

Reply:
Fair enough, i'll change it.